tests/end2end/fixtures/test_qutewm.py

```diff
@@ -0,0 +1,136 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2016 Daniel Schadt
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for qutewm and the qutewm fixtures."""
+
+from PyQt5.QtWidgets import QMainWindow
+
+import pytest
+
+from qutebrowser.utils import qtutils
+from end2end.fixtures import qutewm
+
+
+@pytest.mark.qutewm
+class TestQuteWM:
+
+    # qtbot.wait is used to wait for WAIT_DELAY ms. This makes sure that Qt
+    # processed all events (like the focus changed event from the X server),
+    # so that QMainWindow.isActiveWindow works as expected and returns the
+    # correct result. Waiting for qutewm.window_focused is not enough, as that
+    # may fire when Qt hasn't yet registered the change, so we need to give it
+    # some time and at least enter the event loop once.
+    WAIT_DELAY = 200
```

Comment:
Probably nicer to do this the same way `qutebrowser` does it, i.e. connect to `qapp.focusChanged` and do something like

```python
window = new.window()
if (old is None or old.window() != window) and isinstance(window, OurTestWindow):
    window.window_activated.emit()
```